tests/Owin.Security.OpenIdConnect.Server.Tests/OpenIdConnectAuthorizationRequiredParameterTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.Owin.Testing;
+using Xunit;
+
+namespace Owin.Security.OpenIdConnect.Server.Tests {
+    /// <summary>
+    ///     This class tests a bunch of wrong combination calling Authorization Endpoint,
+    ///     source of truth is http://openid.net/specs/openid-connect-core-1_0.html#AuthorizationEndpoint
+    /// </summary>
+    public class OpenIdConnectAuthorizationRequiredParameterTests {
+        public static TheoryData<HttpMethod> AuthorizationEndpointHttpMethods {
+            get { return new TheoryData<HttpMethod> {HttpMethod.Get, HttpMethod.Post}; }
+        }
+
+        [Theory]
+        [MemberData("AuthorizationEndpointHttpMethods")]
+        public async Task GetAuthorizationEndpoint_NoOpenIdScope_BadRequest(HttpMethod method) {
+            var server = TestServer.Create(app =>
+                app.UseOpenIdConnectServer(options => options.AllowInsecureHttp = true));
+
+            var response = await server.SendMessageAsync(new OpenIdConnectMessage {
+                RequestType = OpenIdConnectRequestType.AuthenticationRequest,
+                AuthorizationEndpoint = "connect/authorize"
+            }, method);
+
+            var openIdResponseMessage = await response.Content.ReadAsOpenIdConnectMessageAsync();
+
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            Assert.False(string.IsNullOrWhiteSpace(openIdResponseMessage.Error));
```

Comment:
But isn't string a list of chars? :smile: 

That said, it has no special case for string and will allocate a new iterator... which is probably less efficient than `Assert.False` + `string.IsNullOrWhiteSpace`.